src/arbitrage.py

```diff
@@ -7,15 +10,16 @@ def __init__(self, limit=0, stock_names=None, api=None):
 
     def pull(self, stock_name):
         """Connects to API to pull data for a given stock and returns a StockData object."""
-        return self.api.get(stock_name)
+        return self.api.get_stock_data(stock_name)
 
     def check(self, expiry_dates=None, option_prices=None, underlying_cost=None):
         """Checks to see if there exists an expiry date for which
         the arbitrage formula for options holds"""
         result = False
         for expiry in expiry_dates:
-            diff_cost = abs(option_prices[expiry][0] - option_prices[expiry][1])
-            if diff_cost / underlying_cost > self.limit:
+            margin = abs(option_prices[expiry]["strike"] - underlying_cost)
+            diff_cost = abs(option_prices[expiry]["call"] - option_prices[expiry]["put"])
+            if diff_cost / underlying_cost > self.limit and diff_cost > margin:
```

Comment:
should be (diff_cost - margin) / underlying_cost > self.limit

are you sure you can just abs the difference like this?

Reply:
can you add a test for this case?

Reply:
resolved